libkbfs/bcache_test.go

```diff
@@ -387,17 +341,38 @@ func TestPutNoHashCalculation(t *testing.T) {
 	// this is an invalid hash; if Put() does not calculate hash, it should go
 	// into the cache
 	block.hash = &kbfshash.RawDefaultHash{}
-	if err := bcache.Put(ptr, tlf, block, TransientEntry); err != nil {
-		t.Errorf("Got error on Put for block %s: %v", ptr.ID, err)
-	}
+	err := bcache.Put(ptr, tlf, block, TransientEntry)
+	require.NoError(t, err)
 
 	// CheckForKnownPtr() calculates hash, which results in a valid hash at
 	// block.hash. If the block with invalid hash was put into cache, this should
 	// fail to find the block.
 	checkedPtr, err := bcache.CheckForKnownPtr(tlf, block)
-	if err != nil {
-		t.Errorf("Unexpected error checking id: %v", err)
-	} else if checkedPtr == ptr {
-		t.Errorf("Put() is calculating hash")
-	}
+	require.NoError(t, err)
+	require.NotEqual(t, ptr, checkedPtr, "Put() is calculating hash")
+}
+
+func TestBlockCacheDoublePut(t *testing.T) {
+	ctx := context.Background()
+	config := blockCacheTestBlockCacheInit(t, 1, 1<<30)
+	defer CheckConfigAndShutdown(ctx, t, config)
+	cache := config.BlockCache().(*BlockCacheStandard)
+	id1 := kbfsblock.FakeID(1)
+	id2 := kbfsblock.FakeID(2)
+	block := makeFakeFileBlock(t, false)
+	bytes := uint64(len(block.Contents))
+
+	t.Log("Put a block into the cache. Check that the cache calculated its byte usage correctly.")
+	testBcachePutWithBlock(t, id1, cache, TransientEntry, block)
+	require.Equal(t, bytes, cache.cleanTotalBytes)
+
+	t.Log("Put the same block into the cache. Check that the cache's byte usage hasn't changed.")
+	testBcachePutWithBlock(t, id1, cache, TransientEntry, block)
+	require.Equal(t, bytes, cache.cleanTotalBytes)
+
+	t.Log("Put a new block into the cache, evicting the first one. Check that the byte usage is updated correctly.")
+	block = NewFileBlock().(*FileBlock)
+	bytes = uint64(len(block.Contents))
```

Comment:
Isn't this always 0? Seems like the test would be a little stronger if it has a non-zero and non-16 number of bytes.